docs/spec/relayer/PacketHandlers.tla

```diff
@@ -0,0 +1,123 @@
+--------------------------- MODULE PacketHandlers ---------------------------
+
+(***************************************************************************
+ This module contains definitions of operators that are used to handle
+ packet datagrams
+ ***************************************************************************)
+
+EXTENDS Integers, FiniteSets, RelayerDefinitions    
+
+(***************************************************************************
+ Packet datagram handlers
+ ***************************************************************************)
+
+\* Handle "PacketRecv" datagrams
+HandlePacketRecv(chainID, chain, packetDatagram, log) ==
+    \* get chainID's connection end
+    LET connectionEnd == chain.connectionEnd IN
+    \* get chainID's channel end
+    LET channelEnd == chain.connectionEnd.channelEnd IN
+    \* get packet
+    LET packet == packetDatagram.packet IN
+    
+    IF \* if the channel and connection ends are open for packet transmission
+       /\ channelEnd.state /= "UNINIT"
+       /\ channelEnd.state = "OPEN"
+       /\ connectionEnd.state /= "UNINIT"
+       /\ connectionEnd.state = "OPEN" 
+       \* if the packet has not passed the timeout height
+       /\ \/ packet.timeoutHeight = 0 
+          \/ chain.height < packet.timeoutHeight  
+       \* if the "PacketRecv" datagram can be verified 
+       /\ packetDatagram.srcChannelID = GetCounterpartyChannelID(chainID)
```

Comment:
This should be checked against channelEnd object, and not from "global" knowledge.